federationapi/routing/send.go

```diff
@@ -185,7 +186,11 @@ func checkAllowedByState(e gomatrixserverlib.Event, stateEvents []gomatrixserver
 	return gomatrixserverlib.Allowed(e, &authUsingState)
 }
 
-func (t *txnReq) processEventWithMissingState(e gomatrixserverlib.Event) error {
+func (t *txnReq) processEventWithMissingState(e gomatrixserverlib.Event, depth int) error {
```

Comment:
Please don't use the term `depth` as I immediately think of event depth, which this is not. Hops/count etc would be better.

Reply:
I've removed the recursion limit for now as @erikjohnston suggests Synapse doesn't have one and that adding one might be a failure mode for a room.